src/plugins/unified_search/public/search_bar/search_bar.tsx

```diff
@@ -42,572 +42,426 @@ export interface SearchBarInjectedDeps {
+  showFilterBar?: boolean;
+  showQueryMenu?: boolean;
+  showQueryInput?: boolean;
+  showSaveQuery?: boolean;
+  showSubmitButton?: boolean;
+  submitButtonStyle?: QueryBarTopRowProps['submitButtonStyle'];
+  // defines size of suggestions query popover
+  suggestionsSize?: SuggestionsListSize;
+  textBasedLanguageModeErrors?: Error[];
+  // User has cleared the active query, your app should clear the entire query bar
+  onClearSavedQuery?: () => void;
   onQueryChange?: (payload: { dateRange: TimeRange; query?: QT | Query }) => void;
   onQuerySubmit?: (
     payload: { dateRange: TimeRange; query?: QT | Query },
     isUpdate?: boolean
   ) => void;
+  onRefresh?: (payload: { dateRange: TimeRange }) => void;
   // User has saved the current state as a saved query
   onSaved?: (savedQuery: SavedQuery) => void;
   // User has modified the saved query, your app should persist the update
   onSavedQueryUpdated?: (savedQuery: SavedQuery) => void;
-  // User has cleared the active query, your app should clear the entire query bar
-  onClearSavedQuery?: () => void;
-
-  onRefresh?: (payload: { dateRange: TimeRange }) => void;
-  indicateNoData?: boolean;
-
-  placeholder?: string;
-  isClearable?: boolean;
-  iconType?: EuiIconProps['type'];
-  nonKqlMode?: 'lucene' | 'text';
-  // defines padding and border; use 'inPage' to avoid any padding or border;
-  // use 'detached' if the searchBar appears at the very top of the view, without any wrapper
-  displayStyle?: 'inPage' | 'detached';
-  // super update button background fill control
-  fillSubmitButton?: boolean;
-  dataViewPickerComponentProps?: DataViewPickerProps;
-  textBasedLanguageModeErrors?: Error[];
   onTextBasedSavedAndExit?: ({ onSave }: OnSaveTextLanguageQueryProps) => void;
-  showSubmitButton?: boolean;
-  submitButtonStyle?: QueryBarTopRowProps['submitButtonStyle'];
-  // defines size of suggestions query popover
-  suggestionsSize?: SuggestionsListSize;
-  isScreenshotMode?: boolean;
-
-  /**
-   * Disables all inputs and interactive elements,
-   */
-  isDisabled?: boolean;
 }
 
 export type SearchBarProps<QT extends Query | AggregateQuery = Query> = SearchBarOwnProps<QT> &
   SearchBarInjectedDeps;
 
-interface State<QT extends Query | AggregateQuery = Query> {
-  isFiltersVisible: boolean;
-  openQueryBarMenu: boolean;
-  showSavedQueryPopover: boolean;
-  currentProps?: SearchBarProps;
-  query?: QT | Query;
-  dateRangeFrom: string;
-  dateRangeTo: string;
-}
-
-class SearchBarUI<QT extends (Query | AggregateQuery) | Query = Query> extends Component<
-  SearchBarProps<QT> & WithEuiThemeProps,
-  State<QT | Query>
-> {
-  public static defaultProps = {
-    showQueryMenu: true,
-    showFilterBar: true,
-    showDatePicker: true,
-    showSubmitButton: true,
-    showAutoRefreshOnly: false,
-  };
-
-  private services = this.props.kibana.services;
-  private savedQueryService = this.services.data.query.savedQueries;
-
-  public static getDerivedStateFromProps(
```

Comment:
💬 I wasn't sure whether this `getDerivedStateFromProps` was still needed, or if it was more of a bandaid needed to cover older React APIs. What do you think, is there something useful in here that we need to keep?

Reply:
I have my answer ;)